src/lib/operators/sort.cpp

```diff
@@ -215,16 +216,12 @@ class Sort::SortImpl : public AbstractReadOnlyOperatorImpl {
 
       auto base_segment = chunk->get_segment(_column_id);
 
-      resolve_segment_type<SortColumnType>(*base_segment, [&](auto& typed_segment) {
-        auto iterable = create_iterable_from_segment<SortColumnType>(typed_segment);
-
-        iterable.for_each([&](const auto& value) {
-          if (value.is_null()) {
-            null_value_rows.emplace_back(RowID{chunk_id, value.chunk_offset()}, SortColumnType{});
-          } else {
-            row_id_value_vector.emplace_back(RowID{chunk_id, value.chunk_offset()}, value.value());
-          }
-        });
+      segment_for_each<SortColumnType>(*base_segment, [&](const auto& value) {
+        if (value.is_null()) {
+          null_value_rows.emplace_back(RowID{chunk_id, value.chunk_offset()}, SortColumnType{});
+        } else {
+          row_id_value_vector.emplace_back(RowID{chunk_id, value.chunk_offset()}, value.value());
```

Comment:
I don't really like the `value.value()` part. We use this in other places as well. In the scan, I have called it `iterator_value`, which I am not very happy with either. Provided that we can come up with a better name for this, we should open an issue to make it consistent.

Reply:
I do not have a major problem with `value.value()`. Whatever you want this to be named like, it shall be named.

Reply:
kk, note that this also means that `SegmentIteratorValue` becomes `SegmentIteratorPosition`

Reply:
Decided I like `SegmentPosition` even better.

Reply:
Sounds good.